src/pyqasm/visitor.py

```diff
@@ -559,7 +576,54 @@ def _check_variable_cast_type(
                 span=statement.span,
             )
 
-    def _visit_measurement(  # pylint: disable=too-many-locals
+    def _qubit_register_consolidation(
+        self, unrolled_stmts: list, total_qubits: int
+    ) -> list[qasm3_ast.Statement]:
+        """
+        Consolidate all quantum registers into a single register '__PYQASM_QUBITS__'.
+
+        Args:
+            unrolled_stmts (list): The list of statements to process and modify in-place.
+
+        Raises:
+            ValidationError: If the total number of qubits exceeds the available device qubits,
+                             or if the reserved register '__PYQASM_QUBITS__' is already declared
+                             in the original QASM program.
+        """
+        if total_qubits > self._device_qubits:  # type: ignore
+            raise_qasm3_error(
+                f"Total qubits '({total_qubits})' exceed device qubits '({self._device_qubits})'.",
+            )
+
+        if "__PYQASM_QUBITS__" in self._global_qreg_size_map:
```

Comment:
Do not need checks for qubits and bits separately. The global scope will already contain variables pertaining to each of the registers. See ref - [[1](https://github.com/qBraid/pyqasm/blob/73c2167d3ea77aaa3706c776ca8208b0d88b298e/src/pyqasm/visitor.py#L339)] , [[2](https://github.com/qBraid/pyqasm/blob/73c2167d3ea77aaa3706c776ca8208b0d88b298e/src/pyqasm/visitor.py#L1545)]